src/charm.py

```diff
@@ -228,11 +225,19 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
 
         if not snap_service.present:
             event.add_status(
-                BlockedStatus("snap service is not installed, please check snap service")
+                BlockedStatus(
+                    f"{SNAP_NAME} snap is not installed. "
+                    "Please wait for installation to complete, "
+                    "or manually reinstall the snap if the issue persists."
+                )
             )
         elif not snap_service.is_active():
             event.add_status(
-                BlockedStatus("snap service is not running, please check snap service")
+                BlockedStatus(
```

Comment:
**question (non-blocking):** 

Can you provide the explanation how this BlockStatus can resolved by charm automatically on the description?

Reply:
Sure, done. :)